components/kyma-environment-broker/internal/event/application_broker.go

```diff
@@ -0,0 +1,48 @@
+package event
+
+import (
+	"context"
+	"reflect"
+	"sync"
+)
+
+type Handler = func(ctx context.Context, ev interface{}) error
+
+type Publisher interface {
+	Publish(ctx context.Context, event interface{})
+}
+
+// ApplicationEventBroker implements a simple event broker which allows to send event across the application.
+type ApplicationEventBroker struct {
+	mu sync.Mutex
+
+	handlers map[reflect.Type][]Handler
+}
+
+func NewApplicationEventBroker() *ApplicationEventBroker {
+	return &ApplicationEventBroker{
+		handlers: make(map[reflect.Type][]Handler),
+	}
+}
+
+func (b *ApplicationEventBroker) Publish(ctx context.Context, ev interface{}) {
```

Comment:
IMO using `Publish` method should be non-blocking. Why do we want to wait until all subscribers will handle that event? It's a little bit against that pattern

Reply:
done